src/popper/runner.py

```diff
@@ -230,6 +231,107 @@ def _prepare_environment(self, step, env={}):
             )
         return step_env
 
+    def _get_build_info(self, step, class_name="DockerRunner"):
```

Comment:
this is kind of defeating the purpose of having a class hierarchy. Instead, this method should do those things that are generic to all children, and then, each child can reuse "as is", override partially (e.g. declare the same function, call the parent method and then do something else, or completely override the method).

Reply:
@ivotron I know I'm going to split the PR but can you please take a look at the new changes I made as regards this? I think I have handled the hierarchical issue